src/service.rs

```diff
@@ -1121,3 +1198,39 @@ where
     }
     Ok(results)
 }
+
+pub(crate) async fn handle_view_finished<TYPES: NodeType<Time = ViewNumber>>(
+    view_number: TYPES::Time,
+    solver_api_url: Url,
+    bid_base_url: Url,
+    bid_config: BidConfig,
+) -> Result<(), BuildError>
+where
+    TYPES::Transaction: BuilderTransaction,
+{
+    // We submit a bid three views in advance.
+    let bid_tx = from_bid_config(bid_config, view_number + 3, bid_base_url)?;
+
+    let solver_client = match connect_to_solver_service::<TYPES>(solver_api_url).await {
+        Some(client) => client,
+        None => {
+            return Err(BuildError::Error {
+                message: "Failed to connect to the solver service.".to_string(),
+            });
+        }
+    };
+
+    if let Err(e) = solver_client
+        .post::<()>("submit_bid")
+        .body_json(&bid_tx)
+        .unwrap()
+        .send()
+        .await
+    {
+        return Err(BuildError::Error {
```

Comment:
I don't think we should return an error here, or the builder will exit on any solver failure, that may well be intermittent. I'd say we just log the error here - and maybe add some retry logic, although that can be added in a separate PR.

Reply:
Updated the return type. Also updated `handle_received_txns` as discussed on Zulip.